### DISCUSSION TITLE
rncp: add blocking function to receive

### DISCUSSION THREAD

**Discussion by pizdjuk** [markqvist/Reticulum#324] [Ideas]
Problem: right now it is needed

1) add callback function for link_established
2) do sleep in a loop waiting for callback function triggered

I dont know, if Reticulum API offers such a possibility, but if not, it would be worth to do. Some blocking function, that wait for some maximum time till link is established.

**Comment by markqvist** [comment]
I don't understand exactly what you mean here. Can you describe what you are trying to do, and how it goes wrong?

**Comment by pizdjuk** [comment]
> I don't understand exactly what you mean here. Can you describe what you are trying to do, and how it goes wrong?

In function receive of rncp

https://github.com/markqvist/Reticulum/blob/97f97eb06316346b3bc2173907720154b6c42447/RNS/Utilities/rncp.py#L37

at the and of the function we enter in permanent cycle
https://github.com/markqvist/Reticulum/blob/97f97eb06316346b3bc2173907720154b6c42447/RNS/Utilities/rncp.py#L87

I think, this is not the best method. As I understand a better way could be add some blocking function. It is much more easy as already is: unblocking callbacks. But sometimes (like this one) it is still useful to have a blocking one.
